pkg/logs/tailers/windowsevent/tailer.go

```diff
@@ -89,210 +94,257 @@ func (t *Tailer) Identifier() string {
-		nameString, ok := name.(string)
-		if !ok {
-			continue
+
+		// subscription is running, wait for and get events
+		select {
+		case <-t.stop:
+			return
+		case _, ok := <-t.sub.EventsAvailable():
+			if !ok {
+				break
+			}
+			// events are available, read them
+			for {
+				events, err := t.sub.GetEvents()
+				if err != nil {
+					// error
+					log.Errorf("GetEvents failed: %v", err)
+					t.sub.Stop()
+					t.source.Status.Error(fmt.Errorf("subscription stopped: %v", err))
+					break
+				}
+				if events == nil {
+					// no more events
+					log.Debugf("No more events")
+					break
+				}
+				for _, eventRecord := range events {
+					t.handleEvent(eventRecord.EventRecordHandle)
+					evtapi.EvtCloseRecord(t.evtapi, eventRecord.EventRecordHandle)
+				}
+			}
 		}
-		nameTextMap[nameString] = text
-	}
-	if len(nameTextMap) == 0 {
-		return nil, fmt.Errorf("no field to transform")
 	}
-	return nameTextMap, nil
 }
 
-// extractParsedBinaryData extract the field Event.EventData.Binary and parse it to its string value
-func extractParsedBinaryData(mv mxj.Map) (string, error) {
-	values, err := mv.ValuesForPath(binaryPath)
-	if err != nil || len(values) == 0 {
-		return "", fmt.Errorf("could not find path: %s", binaryPath)
-	}
-	valueString, ok := values[0].(string)
-	if !ok {
-		return "", fmt.Errorf("could not cast binary data to string: %s", err)
+func (t *Tailer) handleEvent(eventRecordHandle evtapi.EventRecordHandle) {
+
+	richEvt := t.enrichEvent(eventRecordHandle)
+
+	err := t.bookmark.Update(eventRecordHandle)
+	if err != nil {
+		log.Warnf("Failed to update bookmark: %v, to event %s", err, richEvt.xmlEvent)
 	}
 
-	decodedString, _ := parseBinaryData(valueString)
+	msg, err := t.toMessage(richEvt)
 	if err != nil {
-		return "", fmt.Errorf("could not decode %s: %s", valueString, err)
+		log.Warnf("Failed to convert xml to json: %s for event %s", err, richEvt.xmlEvent)
+		return
+	}
+
+	// Store bookmark in origin offset so that it is persisted to disk by the auditor registry
+	offset, err := t.bookmark.Render()
+	if err == nil {
+		msg.Origin.Identifier = t.Identifier()
+		msg.Origin.Offset = offset
+		t.sub.SetBookmark(t.bookmark)
+	} else {
+		log.Warnf("Failed to render bookmark: %v for event %s", err, richEvt.xmlEvent)
 	}
 
-	return decodedString, nil
+	t.source.RecordBytes(int64(len(msg.Content)))
+	t.decoder.InputChan <- msg
 }
 
-// parseBinaryData parses the hex string found in the field Event.EventData.Binary to an UTF-8 valid string
-func parseBinaryData(s string) (string, error) {
-	// decoded is an utf-16 array of byte
-	decodedHex, err := decodeHex(s)
+// enrichEvent renders event record fields using (EvtRender, EvtFormatMessage)
+func (t *Tailer) enrichEvent(event evtapi.EventRecordHandle) *richEvent {
+	xmlData, err := t.evtapi.EvtRenderEventXml(event)
 	if err != nil {
-		return "", err
+		log.Warnf("Error rendering xml: %v", err)
+		return nil
 	}
+	xml := windows.UTF16ToString(xmlData)
+	log.Debug(xml)
```

Comment:
Would it make sense to downgrade it to `Trace`?

Reply:
oops, I think that's a leftover from debugging. There's already a `log.Trace` with the XML in `transform.go`